lib/model/actor.js

```diff
@@ -0,0 +1,18 @@
+const { Subclass } = require('../util/util');
+const AsActor = require('./trait/actor');
+const AsActee = require('./trait/actee');
+const Base = require('./base');
+
+
+const actorTypes = { system: 'system', user: 'user', group: 'group', proxy: 'proxy' };
```

Comment:
How are combinations of groups represented? Are they their own user groups? For example, when we implement campaigns such that each form belongs to a campaign and a campaign can have multiple forms, then a single form may be accessible by (1) users with access to the campaign, (2) any individual users given access, and (3) any user groups given access. Would (1) be represented as a user group? And then (1) + (2) + (3) would be combined into its own user group that represents access to the form?

Reply:
1 would be a user group indeed, because fundamentally everything that can be given permissions is an `actor`, whether user or group. 1+2+3 would not be combined; each entity would be given its own set of grants. See the following answer for a bit more.